src/replay.c

```diff
@@ -0,0 +1,147 @@
+#include "replay.h"
+
+// initial valeus
+char *REPLAY_FILENAME_PRE = "replay_";
+char *REPLAY_PLAY_NAME = "...";
+char *REPLAY_MSG = "";
+const char *REPLAY_FILENAME_DEFAULT_PRE = "replay_";
+const char *REPLAY_FILENAME_SUFFIX = ".replay";
+const char *REPLAY_FOLDER = "./replay";
+int RECORDING_TURN_ON = -1;
+int REPLAY_PLAYING = -1;
+
+FILE *file_replay_writer = NULL;
+FILE *file_replay_reader = NULL;
+
+FILE* get_new_file_replay()
+{
+	char *filename = NULL;
+	uint32_t count = 0;
+	do {
+		asprintf(&filename, "%s/%s%d%s", REPLAY_FOLDER, REPLAY_FILENAME_PRE, count, REPLAY_FILENAME_SUFFIX);
+		count++;
+	} while(access(filename, W_OK) == 0);
+
+	return fopen(filename, "wb");
+}
+
+struct command_raw create_raw_cmd(uint32_t hid_state, uint32_t circle_state, uint32_t cstick_state, uint32_t touch_state, uint32_t special_buttons)
+{
+	struct command_raw r;
+	r.hid_state = hid_state;
+	r.circle_state = circle_state;
+	r.cstick_state = cstick_state;
+	r.touch_state = touch_state;
+	r.special_buttons = special_buttons;
+
+	return r;
+}
+
+void save_command_replay(struct command_raw cmd_raw)
```

Comment:
the only place this is used is the function that builds this buffer.. just pass in the buffer it already created